src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Collections/IntervalTree`1.cs

```diff
@@ -352,30 +351,33 @@ private static Node Balance<TIntrospector>(Node node, in TIntrospector introspec
 
     public IEnumerator<T> GetEnumerator()
     {
-        if (root == null)
-        {
-            yield break;
-        }
+        return this == Empty ? SpecializedCollections.EmptyEnumerator<T>() : GetEnumeratorWorker();
 
-        using var _ = ArrayBuilder<(Node? node, bool firstTime)>.GetInstance(out var candidates);
-        candidates.Push((root, firstTime: true));
-        while (candidates.TryPop(out var tuple))
+        IEnumerator<T> GetEnumeratorWorker()
         {
-            var (currentNode, firstTime) = tuple;
-            if (currentNode != null)
+            if (root == null)
```

Comment:
>if (root == null) [](http://example.com/codeflow?start=12&length=17)

not sure if this is common, but this could be pulled outside the worker too

Reply:
good idea!